src/Libraries/Microsoft.Extensions.Caching.Hybrid/Internal/DefaultJsonSerializerFactory.cs

```diff
@@ -1,42 +1,79 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
 using System.Buffers;
 using System.Diagnostics.CodeAnalysis;
 using System.Text.Json;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.Extensions.Caching.Hybrid.Internal;
 
 internal sealed class DefaultJsonSerializerFactory : IHybridCacheSerializerFactory
 {
+    private readonly IServiceProvider _serviceProvider;
+
+    internal static JsonSerializerOptions FieldEnabledJsonOptions { get; } = new() { IncludeFields = true };
+
+    internal JsonSerializerOptions Options { get; }
+
+    public DefaultJsonSerializerFactory(IServiceProvider serviceProvider)
+    {
+        // store the service provider and obtain the default JSON options, keyed by the **open** generic interface type
+        _serviceProvider = serviceProvider;
+
+#pragma warning disable IDE0079 // unnecessary suppression: TFM-dependent
+#pragma warning disable IL2026, IL3050 // AOT bits
+        Options = serviceProvider.GetKeyedService<JsonSerializerOptions>(typeof(IHybridCacheSerializer<>)) ?? JsonSerializerOptions.Default;
```

Comment:
Will definitely work; I wonder if that is easy to discover, and maybe we can provide some definition like `HybridCacheOptions.JsonSerializerOptionsName` and then resolve via `serviceProvider.GetServiceByName(...)` in order to be more explicit.

_(more a question about why this decision is done than actual comment - the simpler the better I guess)_

Reply:
The problem then becomes the per-type aspect; that's why we're keying by the *open* generic type for the global option, and the *closed* generic type for the per-type option. I mentioned in the PR that we may add a discoverable API for this (that just pokes the same things), but that this needs API design/review.